pkg/operator2/datasync/validation_test.go

```diff
@@ -0,0 +1,351 @@
+package datasync
+
+import (
+	"fmt"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func Test_validateSecret(t *testing.T) {
+	tests := []struct {
+		name    string
+		src     sourceData
+		secrets []*corev1.Secret
+		want    []error
+	}{
+		{
+			name: "secret missing",
+			src:  sourceData{Src: "somesecret", Key: corev1.TLSPrivateKeyKey},
+			want: []error{
+				fmt.Errorf("secret \"somesecret\" not found"),
```

Comment:
(No action required) Consider avoiding a dependency on a specific error message as this style of testing is brittle and will have to be maintained in lock-step with the code.